packages/core/src/sharedWishlists/redux/__tests__/reducer.test.js

```diff
@@ -0,0 +1,369 @@
+import * as actionTypes from '../actionTypes';
+import * as fromReducer from '../reducer';
+import reducer, { entitiesMapper } from '../';
+
+let initialState;
+
+describe('shared wishlists reducer', () => {
+  beforeEach(() => {
+    initialState = reducer();
+  });
+
+  describe('reset handling', () => {
+    it('should return the initial state when there are no fields to keep', () => {
+      expect(
+        reducer(undefined, {
+          payload: {},
+          type: actionTypes.RESET_SHARED_WISHLIST_STATE,
+        }),
+      ).toEqual(initialState);
+    });
+
+    it('should only reset to the initial state the fields specified to reset - reducer root (error)', () => {
+      const state = {
+        error: 'Unexpected Error',
+        isLoading: true,
+        result: null,
+      };
+      const expectedState = {
+        ...state,
+        error: initialState.error,
+      };
+
+      expect(
+        reducer(state, {
+          payload: { fieldsToReset: ['error'] },
+          type: actionTypes.RESET_SHARED_WISHLIST_STATE,
+        }),
+      ).toEqual(expectedState);
+    });
+  });
+
+  describe('error() reducer', () => {
+    it('should return the initial state', () => {
+      const state = reducer().error;
+
+      expect(state).toBe(initialState.error);
+      expect(state).toBeNull();
+    });
+
+    it('should handle CREATE_SHARED_WISHLIST_FAILURE action type', () => {
+      const expectedResult = 'foo';
+
+      expect(
+        reducer(undefined, {
```

Comment:
Avoid using `undefined` for these states to make the test closer to how it will work at runtime.

Reply:
All cases updated ✅